### ISSUE TITLE
Dynamically handle non-existent dynamic imports?

### ISSUE THREAD

**Issue by dcodeIO** [guybedford/es-module-shims#280]
Given a hybrid browser and Node.js ES module doing something like

```js
var fs;
if (isNode) {
  fs = await import("fs");
} else {
  fs = ... browser implementation ...
}
```

where the first dynamic import would only ever execute in Node.js (is not bundled), using es-module-shims produces an error upon polyfilling that the `fs` module cannot be found, which differs from what would happen in native implementations.

This condition can be worked around by additionally specifying

```json
    {
      "imports": {
        "library": "url/to/library.js"
      },
      "scopes": {
        "url/to/": {
          "fs": "./topLevelThrow.js"
        }
      }
    }
```

yet I wonder whether es-module-shims could delay the static error so it only becomes visible if `await import("fs")` actually executes? For example, instead of emitting a static error, I could imagine that modules that are not found during polyfilling could be automatically replaced with a piece of code that throws an error upon first use, similar in effect to the workaround. Is this something worth considering? :)

**Comment by guybedford** [comment]
This is definitely a bug, thanks for reporting! Fix in https://github.com/guybedford/es-module-shims/pull/281.

**Comment by guybedford** [comment]
Fix released in 1.5.3..
